File: Offer 60.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<double> dicesProbability(int n) {
        vector<vector<double>> dp(n, vector<double> (n * 6, 0));
        for (int i = 0; i < 6; ++i)
            dp[0][i] = 1.0 / 6.0;
        for (int i = 1; i < n; ++i) {
            for (int j = 0; j < dp[i].size(); ++j) {
                for (int k = 0; k < 6 && k <= j; ++k) {
                    dp[i][j] += dp[i-1][j-k] / 6.0;
                }
            }
        }
        return vector<double>(dp.back().begin() + n - 1, dp.back().end());
    }
};

File: Offer 60.go
package leetcode

func dicesProbability(n int) []float64 {
	dp := make([][]float64, n)
	for i := 0; i < n; i++ {
		dp[i] = make([]float64, n * 6)
	}
	for i:=0; i < 6; i++ {
		dp[0][i] = 1.0 / 6.0
	}
	for i := 1; i < n; i++ {
		for j := 0; j < len(dp[i]); j++ {
			for k := 1; k <= 6 && k <= j; k++ {
				dp[i][j] += dp[i - 1][j - k] / 6.0
			}
		}
	}
	return dp[n-1][:len(dp[n-1]) - n + 1]
}